system/lib/libc/musl/src/locale/newlocale.c

```diff
@@ -7,7 +7,18 @@ locale_t newlocale(int mask, const char *name, locale_t base)
 {
 	if (*name && strcmp(name, "C") && strcmp(name, "POSIX"))
 		return 0;
-	if (!base) base = calloc(1, sizeof *base);
+	if (!base) {
```

Comment:
Yes, libc++ creates exactly one. So hello world with iostream works thanks to this.

Note that musl has similar opts in other places like they avoid malloc for the first several atexits. This is possibly something upstreamable, I think.